app/middleware.go

```diff
@@ -183,3 +194,48 @@ func (nr *NewRelicMiddleware) Serve(next echo.HandlerFunc) echo.HandlerFunc {
 		return nil
 	}
 }
+
+const metricName = "response_time_milliseconds"
+
+// ResponseTimeMetricsMiddleware struct encapsulating DDStatsD
+type ResponseTimeMetricsMiddleware struct {
+	DDStatsD *middleware.DogStatsD
+}
+
+//ResponseTimeMetricsMiddleware is a middleware to measure the response time
+//of a route and send it do StatsD
+func (responseTimeMiddleware ResponseTimeMetricsMiddleware) Serve(next echo.HandlerFunc) echo.HandlerFunc {
+	return func(c echo.Context) error {
+
+		metricTagsMap := make(map[string]interface{})
+		c.Set("metricTagsMap", metricTagsMap)
+
+		startTime := time.Now()
+		result := next(c)
+		status := c.Response().Status()
+		route := c.Path()
+		method := c.Request().Method()
+		caller := c.Get("metricTagsMap")
+		caller = caller.(map[string]interface{})["gameID"]
+
+		timeUsed := time.Since(startTime)
+
+		tags := []string{
+			fmt.Sprintf("route:%s", route),
+			fmt.Sprintf("method:%s", method),
+			fmt.Sprintf("status:%d", status),
+			fmt.Sprintf("caller:%v", caller),
```

Comment:
Since we are not identifying the caller, but the game ID, you can use "gameID" to match what @thiagomachado did [here](https://git.topfreegames.com/game-tools-and-services/chat-vendor-interface/-/merge_requests/211/diffs#b07e17ac84c0b0934178ac91b62e8efa2b74d96b_35_39). Does it make sense?

Reply:
Yes, totally!